### PR TITLE
Fix install / build by using latest LTS node

### LINKED ISSUE #363: make build-node fails on master
## Requirement - what kind of business use case are you trying to solve?

Unable to install / build master.

## Problem - what in Jaeger blocks you from solving the requirement?

`make build-node` fails. The error is on install. The error is the same with `yarn`, and the `yarn` output is easier to make sense:

```
# etc...
# etc...

[4/4] 🔨  Building fresh packages...
[-/4] ⠠ waiting...
[-/4] ⠠ waiting...
[3/4] ⠠ farmhash
error /[redacted]/tmp/jaeger-client-node/node_modules/farmhash: Command failed.
Exit code: 1
Command: node-gyp rebuild
Arguments:
Directory: /[redacted]/tmp/jaeger-client-node/node_modules/farmhash
Output:
CXX(target) Release/obj.target/farmhash/src/upstream/farmhash.o
warning: include path for stdlibc++ headers not found; pass '-stdlib=libc++' on the command line to use the libc++ standard library instead [-Wstdlibcxx-not-found]
In file included from ../src/upstream/farmhash.cc:23:
../src/upstream/farmhash.h:50:10: fatal error: 'utility' file not found
#include <utility>
         ^~~~~~~~~
1 warning and 1 error generated.
make: *** [Release/obj.target/farmhash/src/upstream/farmhash.o] Error 1
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/[redacted]/.nvm/versions/node/v6.14.3/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:276:23)
gyp ERR! stack     at emitTwo (events.js:106:13)
gyp ERR! stack     at ChildProcess.emit (events.js:191:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:219:12)
gyp ERR! System Darwin 18.2.0
gyp ERR! command "/[redacted]/.nvm/versions/node/v6.14.3/bin/node" "/[redacted]/.nvm/versions/node/v6.14.3/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /[redacted]/tmp/jaeger-client-node/node_modules/farmhash
gyp ERR! node -v v6.14.3
```

## Proposal - what do you suggest to solve the problem or improve the existing situation?

Use a newer version of node. (PR is a WIP.)

## Any open questions to address

N/A

### PR DESCRIPTION
**Pull request by tiffon**
## Which problem is this PR solving?

Fix #363.

## Short description of the changes

Use the current LTS version of node (see [releases](https://nodejs.org/en/about/releases/)).

This adds adds a package-lock.json and causes two prettier formatting changes.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,10 +17,11 @@
   "contributors": ["Yuri Shkuro <ys@uber.com>", "Won Jun Jang <wjang@uber.com>"],
   "dependencies": {
     "node-int64": "^0.4.0",
-    "thriftrw": "^3.5.0",
-    "xorshift": "^0.2.0",
     "opentracing": "^0.13.0",
-    "uuid": "^3.2.1"
+    "prom-client": "^11.0.0",
```
Comment (yurishkuro) [inline package.json]: this isn't a required runtime dependency

Comment (everett980) [inline package.json]: I don't see this dependency in use in the PR, is it required by using a different node version?

Comment (tiffon) [reply package.json]: Awesome catch! npm now saves packages, by default!

>npm install saves any specified packages into dependencies by default.
>– man npm

File: src/configuration_env.js [hunk 2]
```diff
@@ -21,18 +21,20 @@ const deprecatedEnvVars = {
...
-        if (process.env[env]) {
-          console.warn(`You are using deprecated env variable ${env}. Use ${deprecatedEnvVars[env]} instead. \nDeprecated env variable will be removed in the next major release (4.x.x)`)
-        }
-      })
+    Object.keys(deprecatedEnvVars).forEach(env => {
+      if (process.env[env]) {
+        console.warn(
+          `You are using deprecated env variable ${env}. Use ${
+            deprecatedEnvVars[env]
+          } instead. \nDeprecated env variable will be removed in the next major release (4.x.x)`
```
Comment (yurishkuro) [inline src/configuration_env.js]: this split doesn't look right

File: package.json (hunk 1 above)
Comment (tiffon) [reply package.json]: The newer version of npm installs by default. This has been removed and the Makefile updated to not save the dependency when it's installed.

File: src/configuration_env.js (hunk 2 above)
Comment (tiffon) [reply src/configuration_env.js]: It's automatic formatting via prettier.

**Pull request merged**